xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/common/LedgerIndex.java

```diff
@@ -34,9 +40,22 @@ public LedgerIndex(String value) {
    * @param value A {@link String} containing either an integer or a shortcut.
    *
    * @return A {@link LedgerIndex} with the given value.
+   *
+   * @throws NullPointerException if value is null
+   *
+   * @throws NumberFormatException if value is an invalid index
    */
-  public static LedgerIndex of(String value) {
-    return new LedgerIndex(value);
+  public static LedgerIndex of(String value)
+  throws NumberFormatException {
+    if (value == null) {
```

Comment:
As a style, this should be: `Objects.requireNonNull(value);`

Reply:
done